modules/phase_field/src/kernels/ADCHSplitConcentration.C

```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADCHSplitConcentration.h"
+
+registerADMooseObject("PhaseFieldApp", ADCHSplitConcentration);
+
+defineADValidParams(
+    ADCHSplitConcentration,
+    ADKernel,
+    params.addClassDescription("Concentration kernel in Split Cahn-Hilliard that solves chemical "
+                               "potential in a weak form");
+    params.addRequiredCoupledVar("chemical_potential_var", "Chemical potential variable");
+    params.addRequiredParam<MaterialPropertyName>("mobility", "Mobility property name"););
+
+template <ComputeStage compute_stage>
+ADCHSplitConcentration<compute_stage>::ADCHSplitConcentration(const InputParameters & parameters)
+  : ADKernel<compute_stage>(parameters),
+    _grad_mu(adCoupledGradient("chemical_potential_var")),
+    _mobility(getADMaterialProperty<Real>("mobility"))
+{
+}
+
+template <ComputeStage compute_stage>
+ADReal
+ADCHSplitConcentration<compute_stage>::computeQpResidual()
+{
+  return _mobility[_qp] * _grad_mu[_qp] * _grad_test[_i][_qp];
```

Comment:
Hm, we have a similar Kernel in `ADSplitCHCRes` let me review this more closely. I'm concerned we are creating parallel ways to do the same thing.

Reply:
I believe `ADSplitCHCRes` is intended to accept the chemical potential as an argument rather than the kernel variable itself.  It includes the following two lines:

`residual += -_w[_qp] * _test[_i][_qp];`
 `residual += _kappa[_qp] * _grad_u[_qp] * _grad_test[_i][_qp];`

So even if you applied `ADSplitCHCRes` with the kernel variable being the chemical potential, you'd get:
residual = -mu * test + kappa * grad_mu * grad_test
This is different than what we're proposing here:
residual = M * grad_mu * grad_test

The non-AD kernel that our kernel is based on is `CHSplitConcentration`, which has:

`const RealVectorValue a = _mobility[_qp] * _grad_mu[_qp];`
 `return _grad_test[_i][_qp] * a;`

This gives the same results as what we're proposing in AD form.  If we are creating a parallel calculation, is this kernel also parallel to `SplitCHCRes`?